packages/forms/src/widgets/DatalistWidget/DatalistWidget.js

```diff
@@ -151,12 +151,16 @@ class DatalistWidget extends React.Component {
 		};
 	}
 
-	onBlur() {
+	onBlur(event) {
 		if (this.props.options &&
-			this.props.options.restricted && this.state.initalItems.indexOf(this.state.value) === -1) {
+			this.props.options.restricted &&
+			this.state.initalItems.indexOf(this.state.value) === -1) {
 			this.resetValue();
 		} else {
-			this.props.onChange(this.state.value);
+			const { value } = event.target;
+			if (value !== this.state.value) {
```

Comment:
This assignation seems usesless assignation. Why didn't you do that like that : `if (event.target.value !== this.state.value)` ?